hack/generator/pkg/astmodel/storage_conversion_function.go

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"fmt"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astbuilder"
+	"github.com/dave/dst"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sort"
+)
+
+// StoragePropertyConversion represents a function that generates the correct AST to convert a single property value
+// Different functions will be used, depending on the types of the properties to be converted.
+// source is an expression for the source value that will be read.
+// destination is an expression the target value that will be written.
+type StoragePropertyConversion func(
+	source dst.Expr, destination dst.Expr, generationContext *CodeGenerationContext) []dst.Stmt
+
+// StorageConversionFunction represents a function that performs conversions for storage versions
+type StorageConversionFunction struct {
+	// Name of this conversion function
```

Comment:
minor: I think these are supposed to start with the name of the property too although not 100% sure here if the rule applies to non-exported fields as well as exported fields.

really minor though so feel free to ignore this if you like.